doc/source/docs/howto_document.rst

```diff
@@ -73,3 +73,219 @@ Please use the numpydoc `formatting standard`_ as shown in their example_
 
 .. _`formatting standard`: https://numpydoc.readthedocs.io/en/latest/format.html
 .. _example: https://numpydoc.readthedocs.io/en/latest/example.html
+
+.. _doc_c_code:
+
+Documenting C/C++ Code
+**********************
+
+Recently, NumPy headed out to use Doxygen_ along with Sphinx due to the urgent need
+to generate C/C++ API reference documentation from comment blocks, especially with
+the most recent expansion in the use of SIMD, and also due to the future reliance on C++.
+Thanks to Breathe_, we were able to bind both worlds together at the lowest cost.
+
+**It takes three steps to complete the documentation process**:
+
+1. Writing the comment blocks
+-----------------------------
+
+Although there is still no commenting style set to follow, the Javadoc
+is more preferable than the others due to the similarities with the current
+existing non-indexed comment blocks.
+
+.. note::
+   If you have never used Doxygen_ before, then maybe you need to take a quick
+   look at `"Documenting the code" <https://www.doxygen.nl/manual/docblocks.html>`__.
+
+This is how Javadoc style looks like::
+
+  /**
+   * Your brief is here.
```

Comment:
```suggestion
   * function(void) changes everything.
```

Reply:
I don't understand the point behind this suggestion, any reason?